### ISSUE TITLE
SA1316 false positive when inferred names enabled but tuple name is explicitly given

### ISSUE THREAD

**Issue by scottmcginness** [DotNetAnalyzers/StyleCopAnalyzers#3133]
In the following class, there is an error highlighted on the tuple element when a project is configured to check inferred tuple element names:
```
public class TestClass
{
    public string Value { get; set; }
    public (string a, string b) TestMethod()
    {
        return (a: "", b: Value);
        //          here: ^^^^^
    }
}
```
That is, with the settings:
```
{
  "settings": {
    "namingRules": {
      "includeInferredTupleElementNames": true,
      "tupleElementNameCasing" : "camelCase"
    }
  }
}
```

A similar effect occurs when using `PascalCase` and the casing of `a`, `b` and `Value` are flipped in the class above.

I would expect the rule not to be violated, since the tuple element name is in fact `b` (or `B` in the inverted case).
